cdap-ui/app/features/hydrator/services/detail/stores/pipeline-detail-nodeconfig-store.js

```diff
@@ -37,7 +37,17 @@ angular.module(PKG.name + '.feature.hydrator')
     this.setState = function(plugin) {
       onPluginChange.call(this, plugin);
     };
+    this.setPlugin = function(plugin) {
+      this.plugin = plugin;
+      var type = this.isSource ? 'source': false;
```

Comment:
@anwar6953 we use ```isSource, isTransform, isSink``` in html templates too. We had individual properties as true or false because it will make templates a tad bit easier to read. We can change it to be ```type``` and comparing it in template too. That would also work. Will change it in the next fix I push.